### DISCUSSION TITLE
How to create TrayIcon programmatically?

### DISCUSSION THREAD

**Discussion by wl2776** [AvaloniaUI/Avalonia#17764] [Q&A] [answered]
Hi all! I have just started using Avalonia UI.

My goal is to enable system tray icon of an application, depending on preferences read from config file on startup.

Currently I've found an [example](https://docs.avaloniaui.net/docs/reference/controls/tray-icon#example) that makes it always enabled.

Could you provide a sample of code in C#, that creates TrayIcon?

**Comment by tobyfirth** [comment, marked as answer]
You can do something like this.
``` csharp
var icons = new TrayIcons
{
	new TrayIcon
	{
		Icon = new WindowIcon(new Bitmap(AssetLoader.Open(new Uri("avares://SmartbinderUI/Assets/avalonia-logo.ico")))),
		Menu = [
			new NativeMenuItem("Settings")
			{
				Menu = new NativeMenu
				{
					new NativeMenuItem("Option 1"),
					new NativeMenuItem("Option 2"),
					new NativeMenuItemSeparator(),
					new NativeMenuItem("Option 3")
				}
			}
		]
	}
};

TrayIcon.SetIcons(Application.Current, icons);
```

And to remove it.

``` csharp
TrayIcon.SetIcons(Application.Current, null);
```
